plugins/fileservices/fileservices.cpp

```diff
@@ -3140,3 +3140,30 @@ FILESERVICES_API void FILESERVICES_CALL fsClearExpireDays(ICodeContext * ctx, co
     else
         throw makeStringExceptionV(0, "ClearExpireDays: Could not find logical file %s", lfn.str());
 }
+
+static CriticalSection  noCommonDefCrit;
+static int noCommonDef = NotFound;
+FILESERVICES_API bool FILESERVICES_CALL fsGetNoCommonDefault()
+{
+    // Returns the boolean value of 'noCommon' property from the Software/Globals if it is defined,
+    // Otherwise it returns with true, the recent default value of 'noCommon'
+#ifdef _CONTAINERIZED
+    return getComponentConfigSP()->getPropBool("@noCommon", true);
+#else
+    CriticalBlock b(noCommonDefCrit);
+    if (NotFound == noCommonDef)
+    {
+        // Default value if it is not defined in environment.xml
+        noCommonDef = true;
+        Owned<IConstEnvironment> daliEnv = openDaliEnvironment();
+        Owned<IPropertyTree> env = getEnvironmentTree(daliEnv);
+
+        if (env.get())
+        {
+            Owned<IPropertyTree> globals = env->getPropTree("Software/Globals");
+            noCommonDef = globals->getPropBool("@noCommon", noCommonDef);
```

Comment:
trivial: as per code in dfurun, it is unecessary to fetch tree into a linked object, and then the fetch the bool from it. It would be simpler/cleaner to be:
```
    noCommonDef = env->getPropBool("@noCommon", noCommonDef);
```

Reply:
Changed